lib/routemaster/middleware/error_handling.rb

```diff
@@ -5,23 +5,22 @@ module Routemaster
   module Middleware
     class ErrorHandling < Faraday::Response::Middleware
       ERRORS_MAPPING = {
-        (400..400) => Errors::InvalidResource,
-        (401..401) => Errors::UnauthorizedResourceAccess,
-        (403..403) => Errors::UnauthorizedResourceAccess,
-        (404..404) => Errors::ResourceNotFound,
-        (409..409) => Errors::ConflictResource,
-        (412..412) => Errors::IncompatibleVersion,
-        (413..413) => Errors::InvalidResource,
-        (429..429) => Errors::ResourceThrottling,
-        (407..500) => Errors::FatalResource
+        400 => Errors::InvalidResource,
+        401 => Errors::UnauthorizedResourceAccess,
+        403 => Errors::UnauthorizedResourceAccess,
+        404 => Errors::ResourceNotFound,
+        409 => Errors::ConflictResource,
+        412 => Errors::IncompatibleVersion,
+        413 => Errors::InvalidResource,
+        422 => Errors::UnprocessableEntity,
+        429 => Errors::ResourceThrottling,
+        500 => Errors::FatalResource
       }.freeze
 
       def on_complete(env)
-        ERRORS_MAPPING.each do |range, error_class|
-          if range.include?(env[:status])
-            raise error_class.new(env)
-          end
-        end
+        error_class = ERRORS_MAPPING[env[:status]]
+
```

Comment:
I thought about that as well, this list won't cover `407..421` and `423..499`. I've never seen such codes being used anywhere but rails might do something in the background for some particular requests that I might not know about. I guess it's better to play on the safe side and add a catch all for the remaining 4xx codes